### PR TITLE
Add option to simulate without prior knowledge

### PR DESCRIPTION
**Pull request by J535D165**
This PR proposes to implement a feature to simulate a systematic review without any prior knowledge.

```sh
% asreview simulate ~/Downloads/synergy_dataset/van_der_Valk_2021.csv -s $(mktemp).asreview --n_prior_included 0 --n_prior_excluded 0
The following records are prior knowledge:


---0---IRRELEVANT-------------------------------------------
Diabetes mellitus and hypercortisolism in a cat

A fourteen-year-old Persian cat was referred because of poorly controlled diabetes mellitus despite insulin and dietary treatment. Clinical signs were severe polydipsia/polyuria (pupd), poor hair coat quality, stomatitis and hind limb weakness. At the time of initial presentation, he was treated with glargine insulin (0,75 IU/kg BID). A low dose dexamethasone suppression test (LDDST) revealed hypercortisolism (HC). The cat was additionally treated with trilostane, and remission of diabetes mellitus was obtained one year later. This case illustrates the importance of diagnosing an underlying cause of poorly controlled diabetes mellitus. Although hypercortisolism is rare in cats, it is important to consider the disease in these cases. The hypercortisolism in this cat was efficiently managed with trilostane, resulting in a good quality of life.


---1---IRRELEVANT-------------------------------------------
Unexpected Endocrine Features and Normal Pigmentation in a Young Adult Patient Carrying a Novel Homozygous Mutation in the POMC Gene

Proopiomelanocortin (POMC) is the precursor to five biologically active peptides, including ACTH produced in the anterior pituitary and alpha-MSH produced in the hypothalamus. Mutations that inactivate the POMC gene have been described in children, causing a pleiotropic syndrome that includes secondary hypocortisolism, severe obesity, and variable changes in skin and hair pigmentation.We describe a female patient of North African ancestry, homozygous for a frameshift mutation in the POMC gene (6922InsC) that impairs the production of all melanocortin peptides, and that is associated with novel clinical features. Repeated clinical investigations from birth to age 18 yr are presented.ACTH deficiency was diagnosed at birth. Hyperphagia and obesity became apparent before 2 yr of age and rapidly progressed [body mass index (BMI) Z-score, +7 sd at 2 yr, +9.7 sd at 13 yr; BMI, 50 kg/m(2) at 18 yr). At puberty, the patient developed alterations in the somatotropic, gonadotropic, and thyroid axes necessitating hormonal replacement. Surprisingly, there were no obvious pigmentary features; neither the hair color nor measurements of skin reflectance distinguished between the patient and unaffected family members. However, chemical analysis of hair pigment revealed increased production of both pheomelanin and eumelanin.Molecular genetic abnormalities of POMC should always be considered in patients with early onset adrenal insufficiency and obesity, even in the presence of normal pigmentation and multiple pituitary hormone anomalies.


---2---IRRELEVANT-------------------------------------------
Psychological Studies of Adult Male Patients with Sexual Infantilism Before and After Androgen Therapy

Excerpt This study explores some psychological implications of hormonal treatment of adult males who have failed to mature sexually. Although effective hormonal replacement for androgen deficiency ...


---3---IRRELEVANT-------------------------------------------
Hair cortisol as a biological marker of chronic stress: Current status, future directions and unanswered questions

The detrimental effects of stress on human health are being increasingly recognized. There is a critical need for the establishment of a biomarker that accurately measures its intensity and course over time. Such a biomarker would allow monitoring of stress, increase understanding of its pathophysiology and may help identify appropriate and successful management strategies. Whereas saliva and urine cortisol capture real-time levels, hair cortisol analysis presents a complementary means of monitoring stress, capturing systemic cortisol exposure over longer periods of time. This novel approach for cortisol quantification is being increasingly used to identify the effects of stress in a variety of pathological situations, from chronic pain to acute myocardial infarctions . Because of its ability to provide a long-term, month-by-month measure of systemic cortisol exposure, hair cortisol analysis is becoming a useful tool, capable of answering clinical questions that could previously not be answered by other tests. In this paper we review the development, current status, limitations and outstanding questions regarding the use of hair cortisol as a biomarker of chronic stress.


---4---RELEVANT---------------------------------------------
In adults with obesity, copeptin is linked with BMI but is not associated with long-term exposure to cortisol and cortisone

Obesity and cardiometabolic diseases are associated with higher long-term glucocorticoid levels, measured as scalp hair cortisol (HairF) and cortisone (HairE). Cardiometabolic diseases have also been associated with copeptin, a stable surrogate marker for the arginine-vasopressin (AVP) system. Since AVP is, together with corticotropin-releasing hormone (CRH) an important regulator of the hypothalamic-pituitary adrenal axis (HPA axis), we hypothesize that AVP contributes to chronic hypercortisolism in obesity.To investigate whether copeptin levels are associated with Higher HairF and HairE levels in obesity.A cross-sectional study in 51 adults with obesity (BMI ≥30 kg/m2).Associations and interactions between copeptin, HairF, HairE, and cardiometabolic parameters were cross-sectionally analyzed.Copeptin was strongly associated with BMI and waist circumference (WC) (rho = 0.364 and 0.530, P = 0.008 and <0.001, respectively), also after correction for confounders. There were no associations between copeptin and HairF or HairE on a continuous or dichotomized scale, despite correction for confounders.In patients with obesity, AVP seems not a major contributor to the frequently observed high cortisol levels. Other factors which stimulate the HPA axis or affect cortisol synthesis or breakdown may be more important than the influence of AVP on long-term glucocorticoid levels in obesity.

Simulation started

Relevant records found: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 89/89 [00:01<00:00, 55.93it/s]
Records labeled       :  88%|████████████████████████████████████████████████████████████████████████████████████████████▌            | 639/725 [00:01<00:00, 402.98it/s]

Simulation finished
```

The results table

```sh

record_id  label classifier query_strategy balance_strategy feature_extraction  training_set               labeling_time notes
0            0      0       None          prior             None               None            -1  2023-11-08 17:07:49.116134  None
1            1      0       None          prior             None               None            -1  2023-11-08 17:07:49.116134  None
2            2      0       None          prior             None               None            -1  2023-11-08 17:07:49.116134  None
3            3      0       None          prior             None               None            -1  2023-11-08 17:07:49.116134  None
4            4      1       None          prior             None               None            -1  2023-11-08 17:07:49.116134  None
..         ...    ...        ...            ...              ...                ...           ...                         ...   ...
634        288      0         nb            max           double              tfidf           634  2023-11-08 17:07:50.729581  None
635        350      0         nb            max           double              tfidf           635  2023-11-08 17:07:50.732139  None
636        318      0         nb            max           double              tfidf           636  2023-11-08 17:07:50.734629  None
637        134      0         nb            max           double              tfidf           637  2023-11-08 17:07:50.737101  None
638         21      1         nb            max           double              tfidf           638  2023-11-08 17:07:50.739560  None

[639 rows x 9 columns]

```

### CODE REVIEW

File: asreview/review/simulate.py [hunk 1]
```diff
@@ -75,6 +75,28 @@ def sample_prior_knowledge(
     return init
 
 
+def naive_prior_knowledge(labels):
+    """Function to sample prelabelled articles.
```
Comment (J535D165) [inline asreview/review/simulate.py:79]:
```suggestion
    """Select top records until the first 0 and 1 are found.
```

**Comment by jteijema** [comment]
_Select top records_ I assume means the first records from the dataset in the order it was provided?

**Comment by J535D165** [comment]
That's a good question. We can implement that as well. I implemented it this way because it would give you more flexibility. You can shuffle it yourself and simulate it as random. However, this is not very practical for repeated simulation on the same dataset.

**Comment by PeterLombaers** [comment]
A nice addition! It seems like a good idea to start with this simple implementation. We can always add more ways of choosing the prior knowledge, like taking random records.

One thing I was thinking was that functions `naive_prior_knlowedge` and `sample_prior_knowledge` both have the same function and are only used by the `ReviewSimulate` class. I think it makes more sense to add them as static methods to the class. We could have `ReviewClass.sample_prior_knowledge` that takes `start_idx` / `n_prior_included` / `n_prior_excluded` / `labels` as input and the sampling method, and then returns the indices of the priors. I don't think that these functions will ever be used outside of the context of a simulation.

**Comment by jteijema** [comment]
10 simulations using the following settings:

```batch
asreview simulate data\synergy-Appenzeller-Herzog_2019.csv -s output\simulation\synergy-Appenzeller-Herzog_2019\state_files\sim_synergy-Appenzeller-Herzog_2019_0.asreview --n_prior_included 0 --n_prior_excluded 0 --seed 255
```

\* 10, of course. Note that the lines below overlap exactly, as expected.

image

What I am still wondering about, what is the actual use-case for this implementation? Is it a fallback for simulations without prior knowledge? Are we recommending users to put their favorite documents at the top of their dataset?

**Comment by J535D165** [comment]
The idea is that a user might not have prior knowledge. In that case, the naive approach would be to start reading from the top.

**Pull request merged**